src/nemo_safe_synthesizer/pii_replacer/fill.py

```
@@ -0,0 +1,847 @@
+        self._used[pick] = True
+        return self._pool[pick]
+
+
+def _constraint_signature(inst: dict):
+    sfv = inst["select_field_values"]
+    sfv_key = tuple(sorted((k, tuple(v)) for k, v in (sfv or {}).items()))
+    return (inst["sex"], sfv_key)
+
+
+def _stable_hash(s: str) -> int:
+    return int(hashlib.md5(s.encode("utf-8")).hexdigest()[:8], 16)
```

Ruff (0.16.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)